### PR TITLE
Added schema linting.

### PR DESCRIPTION
**Pull request by bharney0**
- Added code snippet schema linting.
- Added row schema linting.
- Added column linting.
- Added moniker schema linting.
- Added zone schema linting.
[AB#228818](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/228818)

### CODE REVIEW

File: docs-linting/markdownlint-custom-rules/codesnippet.js [hunk 1]
```diff
@@ -4,6 +4,32 @@
...
+const schemas = require('./schemas');
+const vscode = require('vscode');
+const { default: axios } = require('axios');
+
+// schema linting
+let allowedInterativeTypes;
+let allowedCodeSnippetAttributes;
+let codeSnippetDataResponse;
+
+function loadImageSchema() {
```
Comment (jamarw) [inline docs-linting/markdownlint-custom-rules/codesnippet.js:16]: Hi @bharney0 , should we change the function to loadCodeSnippetSchema or something similar?

File: docs-linting/markdownlint-custom-rules/codesnippet.js [hunk 2]
```diff
@@ -4,6 +4,32 @@
...
+		.get(schemas.CODESNIPPET_SCHEMA)
+		.then(function (response) {
+			codeSnippetDataResponse = response.data;
+			allowedInterativeTypes = Object.values(codeSnippetDataResponse.properties.interactive.enum);
+			allowedCodeSnippetAttributes = Object.keys(codeSnippetDataResponse.properties);
+		})
+		.catch(function () {
+			const errorMessage = detailStrings.failedResponse
+				.replace('NAME', 'image')
+				.replace('URL', schemas.IMAGE_SCHEMA);
```
Comment (jamarw) [inline docs-linting/markdownlint-custom-rules/codesnippet.js:27]: Should IMAGE_SCHEMA be changed to CODESNIPPET_SCHEMA?

File: docs-linting/markdownlint-custom-rules/codesnippet.js [hunk 3]
```diff
@@ -54,16 +79,6 @@ module.exports = {
 									});
 								}
 
-								//do we have language?
```
Comment (lamebrain) [inline docs-linting/markdownlint-custom-rules/codesnippet.js:57]: Can we delete this?

Comment (bharney0) [reply docs-linting/markdownlint-custom-rules/codesnippet.js:57]: We dont need this. The requirement for the linting is not to require language.

**Comment by bharney0** [comment]
> Add unit tests

@lamebrain I added unit tests. However there were some issues with testing the asynchronous calls to the schema. So I tested was could be tested sync. But I couldnt find a way to test the async calls to the schema.

**Pull request merged**